indexer/index.hpp

```diff
@@ -208,29 +208,29 @@ class Index : public MwmSet
 public:
 
   template <typename F>
-  void ForEachInRect(F & f, m2::RectD const & rect, uint32_t scale) const
+  void ForEachInRect(F && f, m2::RectD const & rect, uint32_t scale) const
```

Comment:
This doesn't make a lot of sense since ReadMWMFunctor stores f by l-value reference.